opm/material/fluidsystems/BlackOilFluidSystem.hpp

```diff
@@ -650,7 +725,8 @@ class BlackOil : public BaseFluidSystem<Scalar, BlackOil<Scalar> >
         switch (phaseIdx) {
         case oilPhaseIdx: {
             if (enableDissolvedGas()) {
-                if (fluidState.phaseIsPresent(gasPhaseIdx)) {
+                //if (fluidState.phaseIsPresent(gasPhaseIdx) && FluidSystem::enableRateLimmitedDissolvedGas()) {
+                if (fluidState.phaseIsPresent(gasPhaseIdx) && !(ThisType::enableRateLimmitedDissolvedGas())) {
```

Comment:
I agree that we have to do something here. If you agree that being able to specify the Rs and Rv values for the "saturated" case should do the trick, I'll whip up a patch...